### PR TITLE
JP-2593: Add more keywords to flat schema

### PR DESCRIPTION
**Pull request by nden**
Resolves [JP-2593](https://jira.stsci.edu/browse/JP-2593)

**Description**

This PR adds additional keywords to the flat schema to be used in the CRDS selectors

`readpatt`, `preadpatt`, `band` - for use by MIRI

Checklist
- [ ] Tests
- [ ] Documentation
- [x] Change log
- [x] Milestone
- [x] Label(s)

### CODE REVIEW

File: jwst/datamodels/schemas/flat.schema.yaml [hunk 1]
```diff
@@ -7,6 +7,10 @@ allOf:
 - $ref: subarray.schema
 - $ref: keyword_filter.schema
 - $ref: keyword_pfilter.schema
+- $ref: keyword_grating.schema
```
Comment (hbushouse) [inline jwst/datamodels/schemas/flat.schema.yaml:10]: If grating and pgrating are for NIRSpec, I don't think they belong in this schema. NIRSpec uses the dflat, fflat, and sflat schemas, not the generic "flat" schema.

Comment (hbushouse) [reply jwst/datamodels/schemas/flat.schema.yaml:10]: https://github.com/spacetelescope/jwst/blob/master/jwst/datamodels/schemas/nirspec_flat.schema.yaml#L8

Comment (nden) [reply jwst/datamodels/schemas/flat.schema.yaml]: removed

**Review by hbushouse — APPROVED** [review]

File: jwst/datamodels/schemas/flat.schema.yaml (hunk 1 above)
Comment (nden) [reply jwst/datamodels/schemas/flat.schema.yaml]: And added `band` for MIRI as I see it in the selector

**Pull request merged**